editors/code/src/inlay_hints.ts

```diff
@@ -151,20 +141,15 @@ class HintsUpdater {
         );
     }
 
-    private async queryHints(documentUri: string): Promise<InlayHint[] | null> {
+    private async queryHints(documentUri: string): Promise<ra.InlayHint[] | null> {
         this.pending.get(documentUri)?.cancel();
 
         const tokenSource = new vscode.CancellationTokenSource();
         this.pending.set(documentUri, tokenSource);
 
-        const request: InlayHintsParams = { textDocument: { uri: documentUri } };
+        const request = { textDocument: { uri: documentUri } };
 
-        return sendRequestWithRetry<InlayHint[]>(
-            this.ctx.client,
-            'rust-analyzer/inlayHints',
-            request,
-            tokenSource.token
-        )
+        return sendRequestWithRetry(this.ctx.client, ra.inlayHints, request, tokenSource.token)
             .catch(_ => null)
```

Comment:
Might as well swtich to `client.logFailedRequest` here and elsewhere?

Reply:
There is no need to log failed request here, we just pass the output channel to `client` [constructor](https://github.com/microsoft/vscode-languageserver-node/blob/f425af9de46a0187adb78ec8a46b9b2ce80c5412/client/src/client.ts#L2585).

Repeating `client.logFailedRequest` everywhere would be very annoying. What if we add the function that wraps `sendRequest` and logs on failure or wrap the client with our own `RaClient`?
```ts
class RustAnalyzerClient {
    inner: lc.LanguageClient;

    async sendRequest(...params) {
        return this.inner.sendRequest(...params).catch(err => {
            this.inner.logFailedRequest(...);
            throw err;
        });
    }

    async sendRequestWithRetry(...params) { ... }
  
    get p2c() { return this.inner.protocol2CodeConverter; }
    get c2p() { return this.inner.code2ProtocolConverter; }
    /* stuff */ 
}
```